### PR TITLE
add GetPullRequest() and CreateComment()

### PR DESCRIPTION
**Pull request by kovetskiy**

### CODE REVIEW

File: stash.go [hunk 1]
```diff
@@ -131,7 +133,7 @@ type (
 	}
 
 	PullRequest struct {
-		ID          int    `id:"closed"`
+		ID          int    `json:"id"`
```
Comment (ae6rt) [inline stash.go]: Nice catch.

File: stash.go [hunk 2]
```diff
@@ -641,6 +651,115 @@ func (client Client) GetPullRequests(projectKey, projectSlug, state string) ([]P
...
+
+		var responseCode int
+		responseCode, data, err = consumeResponse(req)
+		if err != nil {
+			return err
+		}
+		if responseCode != http.StatusOK {
+			var reason string = "unhandled reason"
+			switch {
+			case responseCode == http.StatusBadRequest:
```
Comment (ae6rt) [inline stash.go]: In the documentation for this call here https://developer.atlassian.com/static/rest/stash/3.11.3/stash-rest.html#idp145248, this call can return more than one documented 4xx HTTP status code.  I would like to see each documented status code handled here.

File: stash.go [hunk 3]
```diff
@@ -641,6 +651,115 @@ func (client Client) GetPullRequests(projectKey, projectSlug, state string) ([]P
...
+	err := json.Unmarshal(data, &r)
+	if err != nil {
+		return PullRequest{}, err
+	}
+
+	return r, nil
+}
+
+// CreateComment creates a comment for a pull-request.
+func (client Client) CreateComment(
```
Comment (ae6rt) [inline stash.go]: Please join this up so the function signature is all on one line.

File: stash.go [hunk 4]
```diff
@@ -641,6 +651,115 @@ func (client Client) GetPullRequests(projectKey, projectSlug, state string) ([]P
...
+		),
+		bytes.NewBuffer(reqBody),
+	)
+	if err != nil {
+		return Comment{}, err
+	}
+
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-type", "application/json")
+	req.SetBasicAuth(client.userName, client.password)
```
Comment (ae6rt) [inline stash.go]: Can be called anonymously, as per above comment?

File: stash.go (hunk 2 above)
Comment (kovetskiy) [inline stash.go]: Fixed

File: stash.go (hunk 4 above)
Comment (kovetskiy) [inline stash.go]: Nope, there is should be authenticated user with REPO_READ access.

**Comment by kovetskiy** [comment]
@ae6rt I've fixed all of your comments, thanks for the review.

**Comment by ae6rt** [comment]
Thank you for your patience and contribution.

**Pull request merged**